zone/zonedb.cpp

```diff
@@ -3275,109 +3287,106 @@ void ZoneDatabase::DeleteItemRecast(uint32 character_id, uint32 recast_type)
 
 void ZoneDatabase::LoadPetInfo(Client *client)
 {
-
 	// Load current pet and suspended pet
-	PetInfo *petinfo = client->GetPetInfo(0);
-	PetInfo *suspended = client->GetPetInfo(1);
+	auto pet_info           = client->GetPetInfo(PetInfoType::Current);
+	auto suspended_pet_info = client->GetPetInfo(PetInfoType::Suspended);
 
-	memset(petinfo, 0, sizeof(PetInfo));
-	memset(suspended, 0, sizeof(PetInfo));
+	memset(pet_info, 0, sizeof(PetInfo));
+	memset(suspended_pet_info, 0, sizeof(PetInfo));
 
-	std::string query = StringFormat("SELECT `pet`, `petname`, `petpower`, `spell_id`, "
-					 "`hp`, `mana`, `size` , `taunting` FROM `character_pet_info` "
-					 "WHERE `char_id` = %u",
-					 client->CharacterID());
-	auto results = database.QueryDatabase(query);
-	if (!results.Success()) {
+	const auto& info = CharacterPetInfoRepository::GetWhere(
+		database,
+		fmt::format(
+			"`char_id` = {}",
+			client->CharacterID()
+		)
+	);
+
+	if (info.empty()) {
 		return;
 	}
 
-	PetInfo *pi;
-	for (auto& row = results.begin(); row != results.end(); ++row) {
-		uint16 pet = Strings::ToInt(row[0]);
+	PetInfo* c;
```

Comment:
I don't think we should use `c` since `Client *client` gets passed in already and arguably you could be shortening that to `c`

I was reading this as client before I looked up. At least have it be `p` or stay `pi`